test/erlcloud_emr_tests.erl

```diff
@@ -0,0 +1,207 @@
+-define(_f(F), fun() -> F end).
+
+-define(ACCESS_KEY_ID, string:copies("A", 20)).
+-define(SECRET_ACCESS_KEY, string:copies("a", 40)).
+
+%% These values are from the documentation for the RunJobFlow API
+%% http://docs.aws.amazon.com/ElasticMapReduce/latest/API/API_RunJobFlow.html
+
+-define(RUN_JOB_FLOW_INPUT, [{<<"Name">>,<<"Development Job Flow">>},
+ {<<"Instances">>,
+  [{<<"KeepJobFlowAliveWhenNoSteps">>,<<"false">>},
+   {<<"TerminationProtected">>,<<"false">>},
+   {<<"InstanceGroups">>,
+    [[{<<"Name">>,<<"Master Instance Group">>},
+      {<<"InstanceRole">>,<<"MASTER">>},
+      {<<"InstanceCount">>,1},
+      {<<"InstanceType">>,<<"m1.small">>},
+      {<<"Market">>,<<"ON_DEMAND">>}]]}]},
+ {<<"Steps">>,
+  [[{<<"Name">>,<<"Example Streaming Step">>},
+    {<<"ActionOnFailure">>,<<"CANCEL_AND_WAIT">>},
+    {<<"HadoopJarStep">>,
+     [{<<"Jar">>,<<"/home/hadoop/contrib/streaming/hadoop-streaming.jar">>},
+      {<<"Args">>,
+       [<<"-input">>,<<"s3://elasticmapreduce/samples/wordcount/input">>,
+        <<"-output">>,<<"s3://examples-bucket/example-output">>,<<"-mapper">>,
+        <<"s3://elasticmapreduce/samples/wordcount/wordSplitter.py">>,
+        <<"-reducer">>,<<"aggregate">>]}]}]]},
+ {<<"BootstrapActions">>,[]},
+ {<<"VisibleToAllUsers">>,<<"false">>},
+ {<<"NewSupportedProduct">>,[]},
+ {<<"AmiVersion">>,<<"3.8.0">>}]).
+
+-define(RUN_JOB_FLOW_OUTPUT, [{<<"JobFlowId">>,<<"j-ZKIY4CKQRX72">>}]).
+
+%% Example values from
+%% http://docs.aws.amazon.com/ElasticMapReduce/latest/API/API_AddJobFlowSteps.html
+
+-define(JOB_FLOW_ID, <<"j-3TS0OIYO4NFN">>).
+-define(JOB_FLOW_STEPS, [[{<<"Name">>,<<"Example Jar Step">>},
+  {<<"ActionOnFailure">>,<<"CANCEL_AND_WAIT">>},
+  {<<"HadoopJarStep">>,
+   [{<<"Jar">>,
+     <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/cloudburst.jar">>},
+    {<<"Args">>,
+     [<<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/s_suis.br">>,
+      <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/100k.br">>,
+      <<"s3n:\\/\\/examples-bucket\\/cloudburst\\/output">>,<<"36">>,<<"3">>,
+      <<"0">>,<<"1">>,<<"240">>,<<"48">>,<<"24">>,<<"24">>,<<"128">>,
+      <<"16">>]}]}]]).
+
+-define(ADD_JOB_FLOW_INPUT, [{<<"JobFlowId">>,<<"j-3TS0OIYO4NFN">>},
+ {<<"Steps">>,
+  [[{<<"Name">>,<<"Example Jar Step">>},
+    {<<"ActionOnFailure">>,<<"CANCEL_AND_WAIT">>},
+    {<<"HadoopJarStep">>,
+     [{<<"Jar">>,
+       <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/cloudburst.jar">>},
+      {<<"Args">>,
+       [<<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/s_suis.br">>,
+        <<"s3n:\\/\\/elasticmapreduce\\/samples\\/cloudburst\\/input\\/100k.br">>,
+        <<"s3n:\\/\\/examples-bucket\\/cloudburst\\/output">>,<<"36">>,
+        <<"3">>,<<"0">>,<<"1">>,<<"240">>,<<"48">>,<<"24">>,<<"24">>,
+        <<"128">>,<<"16">>]}]}]]}]).
+
+%% AddJobFlowSteps doesn't have an example output format.
```

Comment:
that is wild, it looks like there is a response syntax that includes [StepIds](http://docs.aws.amazon.com/ElasticMapReduce/latest/API/API_AddJobFlowSteps.html#API_AddJobFlowSteps_ResponseSyntax). I guess their docs guy fell asleep at the wheel.